### PR TITLE
CAMS-620 Fixed Header menu issues

### PR DESCRIPTION
**Pull request by fmaddenflx**
# Purpose

Menu was aligned left and User menu did not contain Admin etc.

# Major Changes

CSS and updated the isAdmin function and renamed it to menuNeedsAdmin which is a more accurate name for the function.

# Testing/Validation

Added tests for menuNeedsAdmin function.

## Summary by Sourcery

Fix header menu alignment and Admin menu behavior, refactor dropdown handlers for consistency, and add tests for the updated menuNeedsAdmin logic

Bug Fixes:
- Fix header CSS to use max-width for proper menu alignment
- Ensure Admin menu item is only added when appropriate by renaming and correcting isAdmin logic to menuNeedsAdmin

Enhancements:
- Refactor DropdownMenu event handlers to wrap single-line if statements in blocks for clarity

Tests:
- Add comprehensive tests for menuNeedsAdmin covering feature flags, user roles, null sessions, and duplicate Admin detection

### CODE REVIEW

File: user-interface/src/lib/components/Header.scss [hunk 1]
```diff
@@ -37,7 +37,7 @@
     padding-right: 0;
     flex-flow: row;
     justify-content: space-between;
-    width: 100%;
```
[the remark this answers is not in the document]
Comment (fmaddenflx) [reply user-interface/src/lib/components/Header.scss:40]: Layout across break points looks correct.  Checked only in Chrome however.

File: user-interface/src/lib/components/Header.test.tsx [hunk 2]
```diff
@@ -210,4 +213,123 @@ describe('Header', () => {
       expect(current).toHaveLength(1);
     });
   });
+
+  describe('menuNeedsAdmin', () => {
```
[the remark this answers is not in the document]
Comment (fmaddenflx) [reply user-interface/src/lib/components/Header.test.tsx:217]: Cool.  Good suggestion I think.
